### PR TITLE
Logging exception thrown in c_operation so that it is not hidden. Cap…

### PR DESCRIPTION
**Pull request by dhirst2880**
…turing exception that can be thrown when running the base64 obfuscator. If this exception wasn't caught then no links would actually be generated for the agent

### CODE REVIEW

File: app/objects/c_operation.py [hunk 1]
```diff
@@ -235,8 +236,8 @@ def report(self, output=False, redacted=False):
             await self._cleanup_operation(services)
             await self.close()
             await self._save_new_source(services)
-        except Exception:
-            pass
+        except Exception as e:
+            logging.error(e)
```
Comment (privateducky) [inline app/objects/c_operation.py]: nice

File: app/utility/base_obfuscator.py [hunk 2]
```diff
@@ -6,7 +8,11 @@ class BaseObfuscator(BaseWorld):
...
         agent = self.__getattribute__('agent')
         supported_platforms = self.__getattribute__('supported_platforms')
-        if agent.platform in supported_platforms and link.ability.executor in agent.executors:
-            o = self.__getattribute__(link.ability.executor)
-            return o(link, **kwargs)
+        try:
+            if agent.platform in supported_platforms and link.ability.executor in agent.executors:
+                o = self.__getattribute__(link.ability.executor)
+                return o(link, **kwargs)
+        except Exception:
```
Comment (privateducky) [inline app/utility/base_obfuscator.py]: what type of exception did you find being thrown here?

File: app/utility/base_obfuscator.py [hunk 3]
```diff
@@ -6,7 +8,11 @@ class BaseObfuscator(BaseWorld):
...
+            logging.error("Failed to run BaseObfuscator, retuning default decoded bytes")
```
Comment (privateducky) [inline app/utility/base_obfuscator.py]: use single quotes on strings and fix spelling on returning

File: app/utility/base_obfuscator.py (hunk 2 above)
Comment (dhirst2880) [reply app/utility/base_obfuscator.py]: AttributeError: 'Obfuscation' object has no attribute 'cmd' was the specific one being thrown.

Comment (privateducky) [reply app/utility/base_obfuscator.py]: ah, now that makes sense. the primary executor of your agent must be "cmd" (command prompt). in your except block, do you need to send back the command?

**Comment by dhirst2880** [comment]
The command was already being returned at the bottom of the run method, I
left that in place.

File: app/utility/base_obfuscator.py (hunk 2 above)
Comment (dhirst2880) [reply app/utility/base_obfuscator.py]: Weird -- my comment didn't follow this one. But in essence, the command was already returned at the base of this function anyway. I didn't alter that behavior.

Comment (privateducky) [reply app/utility/base_obfuscator.py]: nice - thanks!

**Pull request merged**